src/ZiraLink.Api.Application/Tools/Tools.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.NetworkInformation;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace ZiraLink.Api.Application.Tools
+{
+    public class Tools : ITools
+    {
+        public async Task<bool> CheckDomainExists(string domainUrl)
+        {
+            HttpClient _client = new HttpClient();
+            try
+            {
+                HttpResponseMessage response = await _client.GetAsync(domainUrl);
+                var bb = response.Content.ReadAsStringAsync();
```

Comment:
Forgot to put await while calling ReadAsStringAsync

Reply:
Done.